diff --git a/zmq.hpp b/zmq.hpp
index d1fc8cd..147f096 100644
--- a/zmq.hpp
+++ b/zmq.hpp
@@ -547,17 +547,17 @@ namespace zmq
         }
 #endif
         virtual void on_monitor_started() {}
-        virtual void on_event_connected(const zmq_event_t &event_, const char* addr_) {}
-        virtual void on_event_connect_delayed(const zmq_event_t &event_, const char* addr_) {}
-        virtual void on_event_connect_retried(const zmq_event_t &event_, const char* addr_) {}
-        virtual void on_event_listening(const zmq_event_t &event_, const char* addr_) {}
-        virtual void on_event_bind_failed(const zmq_event_t &event_, const char* addr_) {}
-        virtual void on_event_accepted(const zmq_event_t &event_, const char* addr_) {}
-        virtual void on_event_accept_failed(const zmq_event_t &event_, const char* addr_) {}
-        virtual void on_event_closed(const zmq_event_t &event_, const char* addr_) {}
-        virtual void on_event_close_failed(const zmq_event_t &event_, const char* addr_) {}
-        virtual void on_event_disconnected(const zmq_event_t &event_, const char* addr_) {}
-        virtual void on_event_unknown(const zmq_event_t &event_, const char* addr_) {}
+        virtual void on_event_connected(const zmq_event_t &event_, const char* addr_) { (void)event_; (void)addr_; }
+        virtual void on_event_connect_delayed(const zmq_event_t &event_, const char* addr_) { (void)event_; (void)addr_; }
+        virtual void on_event_connect_retried(const zmq_event_t &event_, const char* addr_) { (void)event_; (void)addr_; }
+        virtual void on_event_listening(const zmq_event_t &event_, const char* addr_) { (void)event_; (void)addr_; }
+        virtual void on_event_bind_failed(const zmq_event_t &event_, const char* addr_) { (void)event_; (void)addr_; }
+        virtual void on_event_accepted(const zmq_event_t &event_, const char* addr_) { (void)event_; (void)addr_; }
+        virtual void on_event_accept_failed(const zmq_event_t &event_, const char* addr_) { (void)event_; (void)addr_; }
+        virtual void on_event_closed(const zmq_event_t &event_, const char* addr_) { (void)event_; (void)addr_; }
+        virtual void on_event_close_failed(const zmq_event_t &event_, const char* addr_) { (void)event_; (void)addr_; }
+        virtual void on_event_disconnected(const zmq_event_t &event_, const char* addr_) { (void)event_; (void)addr_; }
+        virtual void on_event_unknown(const zmq_event_t &event_, const char* addr_) { (void)event_; (void)addr_; }
     private:
         void* socketPtr;
     };
